pkg/controller/servicemesh/member/controller.go

```diff
@@ -0,0 +1,193 @@
+		Namespace: member.Spec.ControlPlaneRef.Namespace,
+	}
+	memberRoll := &v1.ServiceMeshMemberRoll{}
+	isNewMemberRoll := false
+	err = r.Client.Get(context.TODO(), memberRollKey, memberRoll)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return reconcile.Result{}, err
+		}
+		// MemberRoll doesn't exist, let's create it
+		isNewMemberRoll = true
+		memberRoll = &v1.ServiceMeshMemberRoll{
+			ObjectMeta: v12.ObjectMeta{
+				Name:      common.MemberRollName,
+				Namespace: member.Spec.ControlPlaneRef.Namespace,
+				Annotations: map[string]string{
+					common.CreatedByKey: controllerName,
+				},
+			},
+		}
+	}
+
+	deleted := member.GetDeletionTimestamp() != nil
+	finalizers := sets.NewString(member.Finalizers...)
+	if deleted {
+		if !finalizers.Has(common.FinalizerName) {
+			reqLogger.Info("ServiceMeshMember deleted")
+			return reconcile.Result{}, nil
+		}
+
+		if !isNewMemberRoll {
+			reqLogger.Info("Removing ServiceMeshMember from ServiceMeshMemberRoll")
+			for i, m := range memberRoll.Spec.Members {
+				if m == member.Namespace {
+					memberRoll.Spec.Members = append(memberRoll.Spec.Members[:i], memberRoll.Spec.Members[i+1:]...)
+					break
+				}
+			}
+
+			memberRollCreatedByThisController := memberRoll.Annotations[common.CreatedByKey] == controllerName
+			if len(memberRoll.Spec.Members) == 0 && memberRollCreatedByThisController {
+				err = r.Client.Delete(context.TODO(), memberRoll)
+				if err != nil {
+					return reconcile.Result{}, errors2.Wrapf(err, "Could not delete ServiceMeshMemberRoll %s/%s", memberRoll.Namespace, memberRoll.Name)
+				}
+			} else {
+				err = r.Client.Update(context.TODO(), memberRoll)
+				if err != nil {
+					return reconcile.Result{}, errors2.Wrapf(err, "Could not update ServiceMeshMemberRoll %s/%s", memberRoll.Namespace, memberRoll.Name)
+				}
+			}
+		}
+
+		reqLogger.Info("Removing finalizer from ServiceMeshMember")
+		finalizers.Delete(common.FinalizerName)
+		member.SetFinalizers(finalizers.List())
+		err = r.Client.Update(context.TODO(), member)
+		if err != nil {
+			return reconcile.Result{}, errors2.Wrapf(err, "Could not update ServiceMeshMember %s/%s when removing finalizer", member.Namespace, member.Name)
+		}
+
+		return reconcile.Result{}, nil
+
+	} else if !finalizers.Has(common.FinalizerName) {
+		reqLogger.Info("Adding finalizer to ServiceMeshMember", "finalizer", common.FinalizerName)
+		finalizers.Insert(common.FinalizerName)
+		member.SetFinalizers(finalizers.List())
+		err = r.Client.Update(context.TODO(), member)
+		return reconcile.Result{}, err
+	}
+
+	if !contains(member.Namespace, memberRoll.Spec.Members) {
+		memberRoll.Spec.Members = append(memberRoll.Spec.Members, member.Namespace)
+
+		if isNewMemberRoll {
+			err = r.Client.Create(context.TODO(), memberRoll)
+			if err != nil {
+				if errors.IsNotFound(err) {
+					reqLogger.Info("Could not create ServiceMeshMemberRoll, because the referenced namespace doesn't exist", "namespace", memberRoll.Namespace)
```

Comment:
The member roll controller was listening for namespace events, as opposed to requeuing the resource.

Reply:
Also fixed in one of the new commits. PTAL